src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -766,6 +767,41 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
         final List<MessageEmbed>       embeds      = map(jsonObject, "embeds",      this::createMessageEmbed);
         final List<MessageReaction>    reactions   = map(jsonObject, "reactions",   (obj) -> createMessageReaction(chan, id, obj));
 
+        MessageActivity activity = null;
+
+        if (!jsonObject.isNull("activity"))
+        {
+            JSONObject activityData = jsonObject.getJSONObject("activity");
+            final MessageActivity.ActivityType activityType = MessageActivity.ActivityType.fromId(activityData.getInt("type"));
+            final String partyId = activityData.optString("party_id", null);
+            MessageActivity.Application application = null;
+
+            switch (activityType)
+            {
+                case JOIN:
+                case SPECTATE:
+                case JOIN_REQUEST:
+                    if (!jsonObject.isNull("application"))
+                    {
+                        JSONObject applicationData = jsonObject.getJSONObject("application");
+
+                        final String name = applicationData.getString("name");
+                        final String description = applicationData.getString("description");
+                        final String iconId = applicationData.getString("icon");
+                        final String coverId = applicationData.getString("cover_image");
+                        final long applicationId = applicationData.getLong("id");
+
+                        application = new MessageActivity.Application(name, description, iconId, coverId, applicationId);
+                    }
+                    break;
+                default:
+                    break;
```

Comment:
Log on debug level about unknown activity type

Reply:
Just for being sure: You want me to add a logger intance (as a class field) with EntityBuilder as the name giving class?

Reply:
I have a feeling this switch is more confusing than anything, what exactly is the point of it? Shouldn't you always just check if `application` is present regardless of type? Then you only need to check that the type is not UNKNOWN.